src/hooks/useCountries.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+
+import { fetchAllCountries } from '../services/countries.js';
+
+export function useAllCountries(order, searchByName, setLoadState) {
+  const [countries, setCountries] = useState([]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchCountries = async () => {
+      try {
+        setLoadState('loading');

```

Comment:
Generally, you want to have the loading state to be a boolean that you toggle from true to false to render either your loading spinner/bar/text if the loading state is `true`, and then when it's `false`, that would indicate that your fetch was successful and that you can safely load your JSX that required the data you were fetching. If you would like to discuss this more in-depth, let me know!